src/main/java/io/netty/buffer/api/CompositeBuffer.java

```diff
@@ -649,13 +652,20 @@ public void ensureWritable(int size, boolean allowCompaction) {
                     // Now we have enough space.
                     return;
                 }
+            } else if (bufs.length == 1) {
+                // If we only have a single component buffer, then we can safely compact that in-place.
+                bufs[0].compact();
+                computeBufferOffsets();
+                if (writableBytes() >= size) {
+                    // Now we have enough space.
+                    return;
+                }
             }
         }
 
-        long newSize = capacity() + (long) size;
-        BufferAllocator.checkSize(newSize);
-        int growth = size - writableBytes();
-        Buffer extension = bufs.length == 0? allocator.allocate(growth) : allocator.allocate(growth, order());
+        int growth = Math.max(size - writableBytes(), minimumGrowth);
```

Comment:
What prevents doing this in a default method and calling `ensureWritable(int, boolean)`?

Reply:
This method will have to do this calculation anyway. The correct default for such an overload is a `minimumGrowth` size of 1.

It wasn't obvious to me which overloads were needed; `(int, boolean)`, `(int, int)`… Adding both of those felt like it would inflate the interface more than necessary.